### ISSUE TITLE
A11y support for phone number with asterisk

### ISSUE THREAD

**Issue by stellaseahll** [flutter/flutter#115624]
Internal: b/258878334

Talkback is not reading strings like '***0789' properly when wrapped in SpellOutText.

Expected talkback: asterisk asterisk asterisk zero seven eight nine
Actual talkback:  three asterisk zero seven eight nine

This is affecting a11y talkback since it is unclear whether  three asterisk zero seven eight nine => 3*0789 OR ***0789.

**Comment by huycozy** [comment]
Hi @stellaseahll 
Can you provide a [completed and minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) so that we may verify this? Also, please provide the output of `flutter doctor -v` as well. Thank you!

**Comment by stellaseahll** [comment]
Hi @huycozy, 

This is a sample code:

final string = '+55 ****** 0789';
Semantics(
        attributedLabel: AttributedString(
            string,
            attributes: [
              SpellOutStringAttribute(
                range: TextRange(start: 0, end: string.length),
              )
            ],
          ),
          excludeSemantics: true,
          child: Text('something else'),
        ),
 );

https://user-images.githubusercontent.com/12376780/202696144-72f8c139-be3c-4aed-b9c9-067981a25b43.mp4

**Comment by huycozy** [comment]
@stellaseahll I think this is working as expected as Android native app. 
Please check below demo:

<details>
<summary>Demo (Flutter and Android native app)</summary>


https://user-images.githubusercontent.com/104349824/202961938-99eebd7c-d81d-45b0-a0b2-96944bf546ed.mp4


</details>

<details>
<summary>Sample code</summary>

```dart
import 'dart:ui';

import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  final string = '+55 ****** 0789';

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
        visualDensity: VisualDensity.adaptivePlatformDensity,
      ),
      home: Scaffold(
        body: Center(
          child: Semantics(
            attributedLabel: AttributedString(
              string,
              attributes: [
                SpellOutStringAttribute(
                  range: TextRange(start: 0, end: string.length),
                )
              ],
            ),
            excludeSemantics: true,
            child: Text('something else'),
          ),
        ),
      ),
    );
  }
}

```
</details>

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.3.8, on macOS 13.0 22A380 darwin-x64, locale en-VN)
    • Flutter version 3.3.8 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 52b3dc25f6 (23 hours ago), 2022-11-09 12:09:26 +0800
    • Engine revision 857bd6b74c
    • Dart version 2.18.4
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A400
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] IntelliJ IDEA Community Edition (version 2022.2.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 70.0.5
    • Dart plugin version 222.4167.21

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.73.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.52.0

[✓] Connected device (3 available)
    • iPhone 14 Pro Max (mobile) • 332079AF-E895-4685-910E-7B1E18B0C6B8 • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-0 (simulator)
    • macOS (desktop)            • macos                                • darwin-x64     • macOS 13.0 22A380 darwin-x64
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 107.0.5304.110

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```

```bash
[!] Flutter (Channel master, 3.6.0-6.0.pre.29, on macOS 13.0 22A380 darwin-x64, locale en-VN)
    • Flutter version 3.6.0-6.0.pre.29 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ee9bc784b5 (2 hours ago), 2022-11-20 19:32:24 -0500
    • Engine revision 271461837e
    • Dart version 2.19.0 (build 2.19.0-406.0.dev)
    • DevTools version 2.19.0
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A400
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] IntelliJ IDEA Community Edition (version 2022.2.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 70.0.5
    • Dart plugin version 222.4167.21

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.73.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.52.0

[✓] Connected device (3 available)
    • SM T225 (mobile) • R9JT3004VRJ • android-arm64  • Android 12 (API 31)
    • macOS (desktop)  • macos       • darwin-x64     • macOS 13.0 22A380 darwin-x64
    • Chrome (web)     • chrome      • web-javascript • Google Chrome 107.0.5304.110

[✓] HTTP Host Availability
    • All required HTTP hosts are available

! Doctor found issues in 1 category.

```
</details>

**Comment by chunhtai** [comment]
Yeah, This is native Android behavior and is outside of our control

**Comment by Haozhe123** [comment]
Android has https://developer.android.com/reference/android/text/style/TtsSpan#TYPE_VERBATIM, which means it is possible to support letter by letter pronounciation on Android?

**Comment by chunhtai** [comment]
@Haozhe123 The TYPE_VERVATIM is what is used behind the spellout text.

I realized there is an different type TYPE_TELEPHONE, may be this can help in this case. I need to investigate more.

**Comment by Haozhe123** [comment]
> The TYPE_VERVATIM is what is used behind the spellout text.

I see, so it broke recently? Might be worth fixing that, because TYPE_TELEPHONE seems like it would only work for phone numbers and not other type of spellout strings e.g. abbreviations.

**Comment by chunhtai** [comment]
@Haozhe123 is this a regression? I think the TYPE_VERVATIM is already used in flutter when you set spellOutText. It would be the Android talkback that decides to pronounce it this way.

**Comment by Haozhe123** [comment]
Seems like it is a regression, see internal bug b/258878334. 

>It would be the Android talkback that decides to pronounce it this way.

Right, so I'm suggesting some changes might be needed on Flutter's integration with Android framework. As it is a very common use case to announce words verbatim, there should be a way in Android to do this.

**Comment by chunhtai** [comment]
I believe this is an Android OS issue. I asked in https://yaqs.corp.google.com/eng/q/4797786058009870336 (internal only). Will update if i know more

**Comment by chunhtai** [comment]
I have not got response to this issue yet, but someone did mention that they would expect people to use read by character feature (swipe up) if they want to know how exact the numbers are spelled.

I will keep this open but there isn't much we can do other than wait for android team's fix

**Comment by chunhtai** [comment]
Close as WAI, Android team's response is that user who really want to read the character will use swipe up/down to iterate through the entire string character by character. They think the current behavior is more desirable.
